common/scala/src/main/scala/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -69,7 +69,7 @@ class KafkaConsumerConnector(
     val wakeUpTask = actorSystem.scheduler.scheduleOnce(cfg.sessionTimeoutMs.milliseconds + 1.second)(consumer.wakeup())
 
     try {
-      val response = consumer.poll(duration.toMillis).asScala
+      val response = Await.result(Future(consumer.poll(duration.toMillis).asScala), 30.seconds)
```

Comment:
How does this relate to the wakeup call 4 lines above? The comment states that it's supposed to wake up the consumer in case poll is infinitely blocked. Does that not work as expected?

Reply:
@markusthoemmes, I don't trust `wakeup()` to always work when the invoker machine is under heavy memory pressure. Perhaps this isn't the solution, but an extra safeguard here would be nice.

Reply:
It's very important to note here that a failing Future does **not** cancel the scheduled action. The poll will be hanging around in the background.

Is there a reason why wakeup wouldn't work under memory pressure, where the new timeout would work better?